### ISSUE TITLE
osTicket 1.10.1 (stable) issue with tickets created from web interface by agent

### ISSUE THREAD

**Issue by snaki4** [osTicket/osTicket#4268]
Good afternoon,

We found a strange issue, replies from the customer to the tickets that were created by agent from web interface triggers a new ticket instead of update for the current one.

So, what is happening, I've found that based on the call to

file: class.thread.php

function lookupByEmailHeaders(&$mailinfo, &$seen=false) 
returns part from subject search

return $ticket->getLastMessage(); (which is always empty) 

and that causes that in

 file : class.mailfetch.php 

        if (($entry = ThreadEntry::lookupByEmailHeaders($vars, $seen))
            && ($message = $entry->postEmail($vars))
        ) {
            if (!$message instanceof ThreadEntry)
                // Email has been processed previously
                return $message;
            // NOTE: This might not be a "ticket"
            $ticket = $message->getThread()->getObject();
        }

is not matched, and it always falling into:

        elseif (($ticket=Ticket::create($vars, $errors, 'Email'))) {
            $message = $ticket->getLastMessage();
        }

so, each reply from the customer will trigger a new ticket. Reply within that ticket will be ok, as it's "initiated" from customer (ticket were created based on reply to welcome message)

We do use Amazon SES for outgoing messages, so it won't much any header for sure and it's expected, but I'm wondering why system can't proceed such situations

**Comment by JediKev** [comment]
@snaki4 

I don't think I've seen this happen before..is the ticket being Closed and then when the customer replies it opens a new ticket or?

Cheers.

**Comment by snaki4** [comment]
@JediKev Nope, I can easy reproduce it by creating new ticket as agent from web interface. That will trigger notification to customer that ticket has been created. Reply from customer to that "welcome" email will cause a new ticket, instead of adding correspondence into ticket that were created by agent.

**Comment by JediKev** [comment]
I cannot reproduce this at all...do you have your emails setup correctly?

**Comment by snaki4** [comment]
I think so, there is a screenshot with configuration.
![example_mail_setup](https://user-images.githubusercontent.com/4473018/39604675-741ff970-4f2d-11e8-9868-251ff2d5aecf.png)

**Comment by JediKev** [comment]
Is the issue summary of the new ticket something like “Re: ... [#original_ticket_number]”? Or is it like a completely new Ticket with all new subject and everything?

**Comment by sperows** [comment]
Note: Sounds very similar to http://osticket.com/forum/discussion/89874/email-responses-creating-new-tickets

**Comment by snaki4** [comment]
@JediKev : Yes, it's like a summary of the welcome email and reply. Take a look for picture
![example_new_ticket](https://user-images.githubusercontent.com/4473018/39618368-aba5d1e8-4f83-11e8-9abb-a3ad0a6d8702.png)

**Comment by snaki4** [comment]
One more info, $ticket->getLastMessage(); (from function lookupByEmailHeaders(&$mailinfo, &$seen=false)) returns empty string (maybe false, didn't check) in case of match by subject in our case.

**Comment by JediKev** [comment]
@snaki4 

As you can see from [include/class.thread.php#L1220-L1244](https://github.com/osTicket/osTicket/blob/develop/include/class.thread.php#L1220-L1244) if all else fails it will look it up by the ticket number in the subject. So I have absolutely no idea why it's failing here for you. Are there any errors via Apache/PHP or osTicket System Logs?

Cheers.

**Comment by snaki4** [comment]
@JediKev 
It's goes inside if, but https://github.com/osTicket/osTicket/blob/24dbd3b3da1b48334abe881347cef1ef23c7d83f/include/class.thread.php#L1242 returns nothing, i.e. empty. 

So, basically, within that part of the code : https://github.com/osTicket/osTicket/blob/24dbd3b3da1b48334abe881347cef1ef23c7d83f/include/class.mailfetch.php#L774-L798 we are falling into new ticket leg.

Here is example.

I have added code to make sure it matches subject:

            if ($ticket->getUserId() == $user->getId() //owner
                    ||  ($c = Collaborator::lookup( // check if collaborator
                            array('user_id' => $user->getId(),
                                  'thread_id' => $ticket->getThreadId())))) {

                $mailinfo['userId'] = $user->getId();
                shell_exec("echo 'matched abc : " . $ticket->getLastMessage() . "' >> /tmp/test");
                return $ticket->getLastMessage();
            }


I got following in /tmp/test:

matched abc : 

So, for some reason it can't return something to the mailfetch class and that causing that new ticket created instead of using old one.

**Comment by sperows** [comment]
@JediKev, sorry for the silly question, but does this apply to collaborators too?

> The only way it would fail at that point is if you don't have an existing ticket with that number (which you should) or it can't verify the user via the email address. When you create a ticket as an Agent you fill out the User Form containing the information of who the ticket is created for. This will add a Guest User account for them that adds a record to the ost_user_email table containing the email address you entered. In include/class.thread.php#L1220-L1244 it **checks for the ticket number AND the user's email. So if the user replied to the ticket with an email address other than the one you entered as an agent then the system will open a new ticket.** So I would check here to make sure this is not happening.

**Comment by JediKev** [comment]
@snaki4 

Is there an Initial Message from the User attached to the original ticket?

Cheers.

**Comment by snaki4** [comment]
@JediKev : nope, it creates another ticket with reply to the "notification of ticket creation"

**Comment by JohnDelisle** [comment]
Anyone aware of a fix being released for this?  

I'm experiencing the same problem since moving to Amazon SES, which replaces the message-id: header with their own values.

What a mess this is creating... so many replies resulting in so many new tickets.

**Comment by JohnDelisle** [comment]
Some more info that may be helpful - 

I added some debugging statements, and can confirm that the code does get to here: https://github.com/osTicket/osTicket/blob/develop/include/class.thread.php#L1242

And it always seems to return null.    In order to get there, the ticket number, user ID, etc. are all tested and confirmed valid.  For whatever reason, `$ticket->getLastMessage()` is null.

In my tests, I've tried replying by email to to a new ticket notification, and replying to an update on a ticket by email.  Both result in passing the tests to get to like 1242, and a null value being returned.

**Comment by JohnDelisle** [comment]
Another debugging note - 

After moving off SES (i.e. delivering directly from my local MTA/relay, not using SES to deliver) the issue is eliminated.  It works just fine.  

It appears the problem is only introduced when the code relies on finding the ticket number via the subject line. 

Any help would be very appreciated.  If there's any debugging I can do, just let me know.
